onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -42,21 +43,31 @@ void DropQDQNodesRules(SelectorActionRegistry& qdq_selector_action_registry) {
       MoveToSlot(dq, ArgType::kInput, 0, ArgType::kInput, 0),
       MoveToSlot(q, ArgType::kOutput, 0, ArgType::kOutput, 0)};
 
-  std::unique_ptr<Action> action = std::make_unique<MergeIntoTarget>(std::move(moves));
-
 #if !defined(ORT_MINIMAL_BUILD)
+  // Use a separate selector + action that disallows 16-bit types for MaxPool and Resize.
+  // int16 MaxPool is not supported by the ONNX specification.
+  // int16 Resize is not supported by the ORT implementation (although allowed by ONNX).
+  std::unique_ptr<NodeSelector> selector_disallow_16bit = std::make_unique<QDQ::DropQDQNodesSelector>(false);
+  std::unique_ptr<Action> action_disallow_16bit = std::make_unique<MergeIntoTarget>(
+      std::vector<NodeAndMoveInfo>(moves));  // Copy before std::move(moves)
+  qdq_selector_action_registry.RegisterSelectorAndAction("drop_no_int16_support",
```

Comment:
the registered names in the selector action registry need to be consistent between a minimal and full build. in a minimal build, the name (saved in the ORT format model) will be used to look up the associated action to run.

for a particular name, we should register both selector and action in a full build and only the action in a minimal build.

Reply:
I added code to register the new action in minimal builds.